js/tsconfig.json

```diff
@@ -1,21 +1,11 @@
 {
-  "include": ["src/**/*.ts", "src/**/*.tsx"],
+  // Use Flarum's tsconfig as a starting point
+  "extends": "flarum-tsconfig",
+  // This will match all .ts, .tsx, .d.ts, .js, .jsx files
+  "include": ["src/**/*"],
   "files": ["shims.d.ts"],
   "compilerOptions": {
-    "allowUmdGlobalAccess": true,
-    "outDir": "dist",
-    "sourceMap": true,
-    "strict": true,
-    "noImplicitReturns": false,
-    "noImplicitAny": false,
-    "module": "es2015",
-    "emitDecoratorMetadata": true,
-    "experimentalDecorators": true,
-    "esModuleInterop": true,
-    "moduleResolution": "node",
-    "target": "es6",
-    "jsx": "preserve",
-    "lib": ["es2015", "es2017", "dom"],
-    "allowSyntheticDefaultImports": true
+    // This will output typings to `dist-typings`
+    "declarationDir": "./dist-typings"
```

Comment:
Do we still want this even though we have the typings repo?

Reply:
It's mainly for extensions to have their typings in a standardised place.

We discussed moving typings to the core repo -- do you want to do that?

Reply:
If possible, I think we should try. Not having to download typings as a separate package, and us not having to update / publish it separately should make things a lot simpler and easier. And that's always good.

Reply:
Totally! It might be worth keeping both, though. That way people can dev outside of a local Flarum install. I've had to do that a few times every now and again.

Reply:
Sticking this here: https://github.com/marketplace/actions/push-directory-to-another-repository